EREGCSC-352 First draft of search

whytheplatypus opened the pull request:
Implements a site wide search feature.

whytheplatypus committed:
First draft of search

- Update links and toc to default to subpart view

whytheplatypus committed:
Add styling to search results page

- include total results
- include link to search in eCFR beta

whytheplatypus committed:
fixup

whytheplatypus committed:
fixup

whytheplatypus commented on regulations/static/regulations/css/scss/partials/_sidebar_right.scss:
```diff
@@ -3,4 +3,18 @@
+
+.right-sidebar {
+    max-width: 400px;
+    div {
+        padding-left: 2em;
+        h4 {
+            margin: 0;
+        }
+        p {
+            color: $secondary_text_color;
+            margin: 0;
+        }
+    }
```
label these more explicitly

whytheplatypus commented on regulations/views/reader.py:
```diff
@@ -91,11 +93,23 @@ def get_view_links(self, context, toc):
 
 class SectionReaderView(ReaderView):
+    def get(self, request, *args, **kwargs):
+        if request.GET.get("section-only"):
+            return super().get(request, *args, **kwargs)
```
Change behavior to redirect to subpart with _part_ view as the default

whytheplatypus committed:
Update section redirect behavior

Redirects to part view as a fallback for the subpart view

whytheplatypus committed:
fixup

cgodwin1 commented on regulations/urls.py:
```diff
@@ -14,7 +15,11 @@
     path('', HomepageView.as_view(), name='homepage'),
     path('<numeric:part>/', RegulationLandingView.as_view(), name="regulation_landing_view"),
     path('<numeric:part>/<version:version>/', PartReaderView.as_view(), name='part_reader_view'),
-    path('<numeric:part>/<numeric:section>/<version:version>/', SectionReaderView.as_view(), name='section_reader_view'),
     path('<numeric:part>/<subpart:subpart>/<version:version>/', SubpartReaderView.as_view(), name="subpart_reader_view"),
+    path('<numeric:part>/<numeric:section>/<version:version>/', SectionReaderView.as_view(), name='section_reader_view'),
```
Should be able to remove these. `section_reader_view` is referenced within:

- content/regulations_landing_pages/landing_433.html
- regulations-site/regulations/views/goto.py
- regulations-site/regulations/views/reader.py
- regulations-site/regulations/tests/urls_test.py
- regulations-site/regulations/templates/regulations/recent-changes.html
- regulations-site/regulations/templatetags/layers.py
- regulations-site/regulations/generator/section_url.py

cgodwin1 commented on regulations/views/utils.py:
```diff
@@ -57,6 +57,10 @@ def find_subpart(section, toc, subpart=None):
+        elif 'Subjgrp' in el['index'] and 'sub_toc' in el:
+            value = find_subpart(section, el['sub_toc'], subpart)
+            if value is not None:
+                return value
```
Could move `if value is not None:` clause outside the conditionals after line 63?

cgodwin1 commented on regulations/views/search.py:
```diff
@@ -0,0 +1,46 @@
+    def get_data(self):
+        return self.accumulate_results([])
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        all_versions = versions.fetch_regulations_and_future_versions()
+        results = self.get_data()
+        context['results'] = list(get_current_results(results, all_versions))
+        return {**context, **self.request.GET.dict()}
```
For consistency, perhaps `get_context_data` should be the first method of the class, as it's immediately obvious to a future reader what is happening

whytheplatypus committed:
Bring in styles for search page

Also begins to remove now unused code paths.
Handle zero state for search page.

whytheplatypus committed:
Updated styles for search page and header

whytheplatypus committed:
Mark matching terms in the search results

whytheplatypus committed:
fixup

whytheplatypus committed:
fixup

whytheplatypus committed:
Secure html generation in templatetag

whytheplatypus committed:
Remove other names for the reader view
